src/app/modules/form/admin-form/__tests__/admin-form.service.spec.ts

```diff
@@ -1191,4 +1193,89 @@ describe('admin-form.service', () => {
       expect(MOCK_UPDATED_FORM.getSettings).toHaveBeenCalledTimes(0)
     })
   })
+
+  describe('deleteFormLogic', () => {
+    const logicId = new ObjectId().toHexString()
+    const mockFormLogic = {
+      form_logics: [
+        {
+          _id: logicId,
+          id: logicId,
+        } as ILogicSchema,
+      ],
+    }
+
+    const UPDATE_SPY = jest
+      .spyOn(FormModel, 'findByIdAndUpdate')
```

Comment:
once you write unit tests for `deleteFormLogic`, you can change this to spy on `deleteFormLogic` instead

Reply:
added spy for deleteFormLogic